### PR TITLE
integrate new pipeline stages for high level data setup

### PR DESCRIPTION
**Pull request by kaanaktas**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CCD-1284

### Change description ###

integrate new pipeline stages for high level data setup

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -113,23 +106,20 @@ withPipeline(type, product, component) {
...
-        }
-        env.DEFINITION_STORE_HOST      = "http://ccd-definition-store-api-${environmentOfDependencies}.service.core-compute-${environmentOfDependencies}.internal"
         env.ELASTIC_SEARCH_FTA_ENABLED = "true"
+    } else {
+        env.DEFINITION_STORE_URL_BASE = "http://ccd-definition-store-api-pr-575.service.core-compute-preview.internal"
     }
 
     echo "ES FTA Enabled = ${env.ELASTIC_SEARCH_FTA_ENABLED} on branch ${env.BRANCH_NAME}"
-    echo "FTA will use Definition Store: ${env.DEFINITION_STORE_HOST}"
+    echo "FTA will use Definition Store: ${env.DEFINITION_STORE_URL_BASE}"
```
Comment (MSancaktutar) [inline Jenkinsfile_CNP:115]: please remove this line. will print null many time.

File: src/test/java/uk/gov/hmcts/ccd/domain/types/CaseDataValidatorTest.java [hunk 2]
```diff
@@ -229,210 +229,6 @@ public void validValueCollection() throws Exception {
...
-            + "        }\n"
-            + "    }";
-
-        final Map<String, JsonNode> values = MAPPER.readValue(DATA, new TypeReference<HashMap<String, JsonNode>>() {
-        });
-        final ValidationContext validationContext = getValidationContextDynamicFields(values);
-        final List<ValidationResult> result = caseDataValidator.validate(validationContext);
-        assertEquals(result.toString(), 0, result.size());
-    }
-
```
Comment (MSancaktutar) [inline src/test/java/uk/gov/hmcts/ccd/domain/types/CaseDataValidatorTest.java:435]: let's not forget to reenable this flaky test

File: Jenkinsfile_CNP (hunk 1 above)
Comment (kaanaktas) [reply Jenkinsfile_CNP]: done

File: src/test/java/uk/gov/hmcts/ccd/domain/types/CaseDataValidatorTest.java (hunk 2 above)
Comment (kaanaktas) [reply src/test/java/uk/gov/hmcts/ccd/domain/types/CaseDataValidatorTest.java]: done

**Review by MSancaktutar — APPROVED** [review]
LGTM

**Review by mattnayler — CHANGES_REQUESTED** [review]
A few minor comments added.  It's good to see the pipeline for this green.

File: src/aat/resources/features/F-048/S-135.td.json [hunk 3]
```diff
@@ -16,7 +16,7 @@
 
 	"user": {
 		"username": "auto.test.cnp@gmail.com",
-		"password": "[[$CCD_IMPORT_AUTOTEST_PASSWORD]]"
+		"password": "[[$DEFINITION_IMPORTER_PASSWORD]]"
```
[the remark this answers is not in the document]
Comment (MSancaktutar) [reply src/aat/resources/features/F-048/S-135.td.json:19]: nice one. leave it at least slightly nicer than found it. :)

[the remark this answers is not in the document]
Comment (kaanaktas) [reply src/aat/resources/features/F-048/S-135.td.json]: done

File: src/aat/resources/features/F-054/S-095.td.json [hunk 4]
```diff
@@ -11,7 +11,7 @@
 
 	"user": {
 		"username": "auto.test.cnp@gmail.com",
-		"password": "[[$CCD_IMPORT_AUTOTEST_PASSWORD]]"
+		"password": "[[$DEFINITION_IMPORTER_PASSWORD]]"
```
[the remark this answers is not in the document]
Comment (kaanaktas) [reply src/aat/resources/features/F-054/S-095.td.json]: done

File: src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreTestAutomationAdapter.java [hunk 5]
```diff
@@ -54,13 +49,8 @@ private synchronized String getDataFileTag(Scenario scenario) {
...
     protected BeftaTestDataLoader buildTestDataLoader() {
-        return new DefaultBeftaTestDataLoader() {
-            @Override
-            public void doLoadTestData() {
-                DataStoreTestAutomationAdapter.this.loader.addCcdRoles();
-                DataStoreTestAutomationAdapter.this.loader.importDefinitions();
-            }
-        };
+        return new DataLoaderToDefinitionStore(this,
+                "uk/gov/hmcts/ccd/test_definitions/valid");
```
[the remark this answers is not in the document]
Comment (kaanaktas) [reply src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreTestAutomationAdapter.java]: done

File: Jenkinsfile_CNP [hunk 6]
```diff
@@ -113,23 +106,19 @@ withPipeline(type, product, component) {
...
     if (env.BRANCH_NAME == 'master' || env.BRANCH_NAME == 'demo' || env.BRANCH_NAME == 'perftest' || env.BRANCH_NAME == 'ithc' || prsToUseAat.toLowerCase().contains(env.BRANCH_NAME.toLowerCase())) {
-        environmentOfDependencies = env.BRANCH_NAME
-        // use aat environment for dependencies of master and CI/CD builds.
-        if (env.BRANCH_NAME.startsWith("PR") || env.BRANCH_NAME == 'master') {
-            environmentOfDependencies = "aat"
-        }
-        env.DEFINITION_STORE_HOST      = "http://ccd-definition-store-api-${environmentOfDependencies}.service.core-compute-${environmentOfDependencies}.internal"
         env.ELASTIC_SEARCH_FTA_ENABLED = "true"
+    } else {
+        env.DEFINITION_STORE_URL_BASE = "http://ccd-definition-store-api-pr-575.service.core-compute-preview.internal"
```
[the remark this answers is not in the document]
Comment (kaanaktas) [reply Jenkinsfile_CNP]: done

**Review by MSancaktutar — APPROVED** [review]
LGTM

**Review by mattnayler — APPROVED** [review]
LGTM 👍

**Pull request merged**